### DISCUSSION TITLE
API 요청 목록

### DISCUSSION THREAD

**Discussion by HyeonsikBae** [innovationacademy-kr/Cabi#145] [FrontEnd]
## API 요청 목록

- [ ] 대여하려는 공유사물함에 대한 정보 요청
  API 요청 시점
  - 대여가 가능한 공유사물함을 누른 경우

```
method: GET
response_data: {
  cabinet_type: string;
  cabinet_id: number;
  location: string;
  floor: number;
  cabinet_num: number;
  cabinet_title: string;
  lent_user_id: number[];
  intra_id: string[];
  lent_time: string;
  expire_time: string;
  isLent: number;
}

```
<br>

- [ ] 대여한 공유사물함에 대한 정보 요청
  API 요청 시점
  - 대여한 공유사물함을 누른 경우
  - 대여한 공유사물함이 있는 상태에서 로그인하는 경우
  - 대여 버튼을 누르고 대여가 완료된 경우

```
method: GET
response_data: {
  cabinet_type: string;
  cabinet_id: number;
  location: string;
  floor: number;
  cabinet_num: number;
  cabinet_title: string;
  cabinet_memo: string;
  lent_user_id: number[];
  intra_id: string[];
  lent_time: string;
  expire_time: string;
  isLent: number;
}

```
<br>

- [ ] title update
  API 요청 시점
  - 대여한 사물함의 방 제목이 변경되는 경우

```
method: POST
request_data: {
  cabinet_id: number;
  cabinet_title: string;
  lent_user_id: number;
  intra_id: string;
  edit_time: Date;
}
```
<br>

- [ ] memo update
  API 요청 시점
  - 대여한 사물함의 메모가 변경되는 경우

```
method: POST
request_data: {
  cabinet_id: number;
  cabinet_memo: string;
  lent_user_id: number;
  intra_id: string;
  edit_time: Date;
}
```
<br>

- [ ] 공유사물함 반납
  API 요청 시점
  - 대여한 사물함을 반납하는 경우

```
method: POST
request_data: {
  cabinet_id: number;
  lent_user_id: number;
  intra_id: string;
  return_time: Date;
}
```
<br>

- [ ] 공유사물함 대여
  API 요청 시점
  - 사물함 대여 버튼을 누르는 경우
```
method: POST
request_data: {
  cabinet_id: number;
  lent_user_id: number;
  intra_id: string;
  lent_time: Date;
}
```

**Comment by joohongpark** [comment]
1. lent_user_id와 intra_id는 각자 배열로 담는 것보다는 별개의 객체로 묶어서 그 객체의 배열로 사용하는 것이 좋을 거 같습니다.
예를 들면
```
interface user_data {
  lent_user_id: number;
  intra_id: string;
}
```
와 같이 정의하고, 다른 객체에서 `user_data[]` 처럼 사용하는 것을 건의 드립니다.
2. title update을 비롯한 업데이트 관련
서버 단에서 토큰을 이용해 사용자를 자동으로 식별하기 때문에 사용자 정보(lent_user_id, intra_id)는 불필요 할 것 같습니다.
그리고 Date 타입은 업데이트 시간을 체크하기 위한 데이터라면 서버 단에서 시간을 체크하기 때문에 해당 필드도 불필요할 것 같습니다.
3. 전반적으로 기존 req, res 데이터 타입 필드명에 대해 리팩터링을 추후에 진행하는 것이 좋을 것 같습니다. (예를 들면 isLent 라던지)
